index.js

```diff
@@ -1,55 +1,88 @@
-const declare = require('@babel/helper-plugin-utils').declare;
 const buildTargets = require('@zapier/browserslist-config-zapier');
+const declare = require('@babel/helper-plugin-utils').declare;
+const invariant = require('invariant');
+
+const compact = xs => xs.filter(Boolean);
+
+const validateTargetOption = (target = 'browser') => {
+  const validTargets = ['browser', 'node'];
+  invariant(
+    validTargets.includes(target),
+    `Invalid option: The target option '${target}' should be either 'browser' or 'node'.`
+  );
+  return target;
+};
+
+const normalizeOptions = (options = {}) => ({
+  target: validateTargetOption(options.target),
+});
 
-module.exports = declare(api => {
+const configurePresets = (env, target) =>
+  compact([
+    [
+      '@babel/preset-env',
+      {
+        modules: env === 'test' || target === 'node' ? 'commonjs' : false,
+        targets:
+          target === 'browser'
+            ? {
+                browsers: buildTargets,
+              }
+            : {
+                node: 'current',
+              },
+      },
+    ],
+    target === 'browser' && '@babel/preset-react',
+    '@babel/preset-flow',
+  ]);
+
+const configurePlugins = (env, target) =>
+  compact([
+    '@babel/plugin-syntax-dynamic-import',
+    '@babel/plugin-proposal-object-rest-spread',
+    '@babel/plugin-proposal-class-properties',
+    // Need to force-transform classes due to an Edge bug. See: https://github.com/zapier/zapier/pull/21319#issuecomment-437871079
+    // Once the fix for this lands on Edge, we can get rid of this (but make sure to test in older versions of Edge first!).
+    target === 'browser' && '@babel/plugin-transform-classes',
```

Comment:
This guy blocked me [here](https://github.com/zapier/zapier/pull/22727). Node supports classes so there is no need to transpile anyway.

Reply:
Do you mean no need to transpile `class`es? (just making sure I understand 😄 )

Reply:
Yessir, Node natively supports class since version 6: https://node.green/

Reply:
Instead of doing all of this, could we check if we can get rid of the `@babel/plugin-transform-classes` plugin entirely? It's been a while since I added that, I think that Edge bug should have been fixed by now. That way we wouldn't need to configure a `target` at all.

Reply:
@vitorbal I can check for that the bug has been resolved and remove it if so.

Can you define "all of this"? Do you mean here or the whole PR? I still believe we would benefit of having the option to select a target.

Reply:
@vitorbal This still happening for Edge 16 and 17, but not in Edge 18 (latest). Our browser list targets the [last 3 major versions of Edge](https://github.com/zapier/browserslist-config-zapier/blob/master/index.js#L5), so it seems we still need this little dude for some time 😞

Reply:
dang :( thanks for checking anyway!